### DISCUSSION TITLE
Why detect side of collision before collision?

### DISCUSSION THREAD

**Discussion by Pictony** [fabricjs/fabric.js#7704] [Q&A]
Hello!

Not a problem but a question about a problem I solved. I don't understand why it works the way it works :-)

The problem was about collision and collision side detections. In my first attempts, I was detecting collisions first, then I was determining which sides of the objects entered in collision. It never worked perfectly, there was always a problem or another, no matter how I detect both of them.
I solved the problem by determining firtsly the positions of each objects relative to each others and then detecting collisions.

It seems not logical to me so I'd like to understand why the second method works and not the first one... Any ideas?

**Comment by ShaMan123** [comment]
Hard to say, you need to share more for anyone to say otherwise.
Are you using a built in functionality?

**Reply by Pictony** [reply]
I tried to detect collision many ways, with built in functionnalities (intersectsWithObject, intersectswithRect) and without (comparisons of positions with conditions). When I was detecting collision first, I detected the side by using the last known position of the object just before collision (_stateProperties).
Finally, it now works by detecting relative positions of each objects fisrt, then detecting collision, both of them only by comparing positions with conditions.
